pandas/core/arrays/arrow/array.py

```diff
@@ -661,6 +667,23 @@ def reshape(self, *args, **kwargs):
             f"as backed by a 1D pyarrow.ChunkedArray."
         )
 
+    @doc(ExtensionArray.searchsorted)
+    def searchsorted(
+        self,
+        value: NumpyValueArrayLike | ExtensionArray,
+        side: Literal["left", "right"] = "left",
+        sorter: NumpySorter = None,
+    ) -> npt.NDArray[np.intp] | np.intp:
+        if self._hasna:
+            raise ValueError(
```

Comment:
Is this the same error the base class would raise? Are there tests for this?

Reply:
No, the base class raises:

```
TypeError: boolean value of NA is ambiguous
```

The `ArrowExtensionArray.searchsorted` method added here is pretty much a copy of `BaseMaskedArray.searchsorted`. I will add a test. (I'll add one for BaseMaskedArray as well - I don't see it being tested there)